cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestBase.java

```diff
@@ -53,29 +59,121 @@
 import org.junit.Before;
 import org.junit.ClassRule;
 import org.junit.rules.TemporaryFolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.io.File;
 import java.io.IOException;
 import java.net.URI;
 import java.util.List;
 import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.Callable;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 import javax.annotation.Nullable;
+import javax.ws.rs.ServiceUnavailableException;
 
 /**
  *
  */
 public class IntegrationTestBase {
+  private static final Logger LOG = LoggerFactory.getLogger(IntegrationTestBase.class);
+  private static final long SERVICE_CHECK_TIMEOUT = TimeUnit.MINUTES.toSeconds(10);
 
   @ClassRule
   public static final TemporaryFolder TEMP_FOLDER = new TemporaryFolder();
 
+  private AccessToken accessToken;
+
   @Before
   public void setUp() throws Exception {
+    checkSystemServices();
+    assertUnrecoverableResetEnabled();
     assertIsClear();
   }
 
+  private void checkSystemServices() throws ServiceUnavailableException {
+
+    Callable<Boolean> monitorCallable = new Callable<Boolean>() {
+      @Override
+      public Boolean call() throws Exception {
+        return getMonitorClient().allSystemServicesOk();
+      }
+    };
+
+    try {
+      checkServicesWithRetry(monitorCallable, "CDAP Services are not available");
+    } catch (Throwable e) {
+      if (e.getCause() instanceof UnauthorizedException) {
+        // security is enabled, we need to get access token before checking system services
+        try {
+          accessToken = fetchAccessToken();
+          checkServicesWithRetry(monitorCallable, "CDAP Services are not available");
+        } catch (Exception ex) {
+          throw Throwables.propagate(ex);
+        }
+      } else {
+        throw Throwables.propagate(e);
+      }
+    }
+    LOG.info("CDAP Services are up and running!");
+  }
+
+  /**
+   * Uses BasicAuthenticationClient to fetch {@link AccessToken} - this implementation can be overridden if desired.
+   * @return {@link AccessToken}
+   * @throws IOException
+   */
+  protected AccessToken fetchAccessToken() throws IOException {
+    Properties properties = new Properties();
+    properties.setProperty("security.auth.client.username", System.getProperty("username"));
```

Comment:
It would be a good idea to call the properties `cdap.username` and `cdap.password`, so that it does not clash with any other username/password being used.